Content/ContentTypeResolver/TeaserSelectionResolver.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\HeadlessBundle\Content\ContentTypeResolver;
+
+use Sulu\Bundle\HeadlessBundle\Content\ContentView;
+use Sulu\Bundle\HeadlessBundle\Content\Serializer\TeaserSerializerInterface;
+use Sulu\Bundle\PageBundle\Teaser\Teaser;
+use Sulu\Bundle\PageBundle\Teaser\TeaserContentType;
+use Sulu\Bundle\PageBundle\Teaser\TeaserManagerInterface;
+use Sulu\Component\Content\Compat\PropertyInterface;
+
+class TeaserSelectionResolver implements ContentTypeResolverInterface
+{
+    public static function getContentType(): string
+    {
+        return 'teaser_selection';
+    }
+
+    /**
+     * @var TeaserManagerInterface
+     */
+    private $teaserManager;
+
+    /**
+     * @var TeaserSerializerInterface
+     */
+    private $teaserSerializer;
+
+    public function __construct(TeaserManagerInterface $teaserManager, TeaserSerializerInterface $teaserSerializer)
+    {
+        $this->teaserManager = $teaserManager;
+        $this->teaserSerializer = $teaserSerializer;
+    }
+
+    public function resolve($data, PropertyInterface $property, string $locale, array $attributes = []): ContentView
+    {
+        $value = $this->getValue($property);
```

Comment:
Data should not be read from the `property` the property should only be accessed to accessed `params` from the xml. All data to load the teaser should come from the `$data` array.

Reply:
```suggestion
```

Reply:
@alexander-schranz I copied those two functions from the `TeaserContentType` with the idea to have the exact same behaviour ... But I'm going to change it